source/Hardpoint.cpp

```diff
@@ -191,7 +192,24 @@ bool Hardpoint::CanAim(const Ship &ship) const
 // Check if this weapon is ready to fire.
 bool Hardpoint::IsReady() const
 {
-	return outfit && burstReload <= 0. && burstCount;
+	return outfit && burstReload <= 0. && burstCount && !IsBlind();
+}
+
+
+
+// Check if this weapon can't fire because of its blindspots.
+bool Hardpoint::IsBlind() const
+{
+	return any_of(baseAttributes.blindspots.begin(), baseAttributes.blindspots.end(),
+		[this](pair<Angle, Angle> blindspot)
+		{
+			if(blindspot.first.Degrees() <= blindspot.second.Degrees())
+				return angle.Degrees() >= (blindspot.first + baseAngle).Degrees()
+					&& angle.Degrees() <= (blindspot.second + baseAngle).Degrees();
+			else
+				return angle.Degrees() >= (blindspot.first + baseAngle).Degrees()
+					|| angle.Degrees() <= (blindspot.second + baseAngle).Degrees();
```

Comment:
My suggestion about `IsInRange` was to use the IsInRange function here. It should do exactly what is required, it has some unit-tests and it should be optimized for angles (so no need to convert to degrees).
```suggestion
			return angle.IsInRange(blindspot.first + baseAngle, blindspot.second + baseAngle);
```